### ISSUE TITLE
Filtering second level child collection

### ISSUE THREAD

**Issue by kad1r** [kahanu/System.Linq.Dynamic#57]
Hi, I don't know it's an issue but I want to filter my second level included child collection. When I try to filter it says "No property or field" error. I try to explain that down below.

> No property or field 'PRODUCTS' exists in type 'ICollection1'`

```
var orders = _uow.Repository<ORDERS>()
    .Query()
    .Where("PRODUCTS.HEADING.ToLower().Contains(\"foo\")")
    //.Where("ORDER_DETAILS.PRODUCTS.HEADING.ToLower().Contains(\"foo\")") - I also tried that but same result
    .Include("ORDER_DETAILS")
    .Include("ORDER_DETAILS.PRODUCTS")
    .ToList();
```

Here is my ef query. I'm using repository with unit of work. Unit of work returns IQueryable. 
I have orders (shopping) and end user can see products of orders in the list. On the list you can search for products.
Is it possible to filter second level child collection?

**Comment by StefH** [comment]
So the uow return a IQueryable<Order> object?

And an Order has a property called Products?

**Comment by kad1r** [comment]
Yea. Order, Order_Details and Products are all my tables.
Order_Details is child collection of Order and Products is child collection of Order_Details too.

**Comment by StefH** [comment]
So an Order has zero or more Order_Details ?

And a single Order_Detail has zero or more Products?

And are you sure that the naming convention is cotrect?

Order_Details should maybe be OrderDetails?

**Comment by kad1r** [comment]
All naming convetions are correct. And for validation every order must have order details and every order detail has products. Otherwise system won't be worked.

**Comment by kad1r** [comment]
I found a solution. And that make sense. I asked the same question before on stackoverflow and someone has replied me the correct answer.
http://stackoverflow.com/questions/38118300/how-to-filter-child-collection-with-linq-dynamic/38311180#38311180

I did not know how to use entity framework any expression with linq dynamic but it was very simple.
`.Where("ORDER_DETAILS.Any(PRODUCTS.HEADING.ToLower().Contains(\"foo\"))")`
